src/site/content/en/blog/webassembly-threads/index.md

```diff
@@ -0,0 +1,472 @@
+results, you can use [`pthread_detach`](https://man7.org/linux/man-pages/man3/pthread_detach.3.html)
+instead of `pthread_join`. This will leave the thread callback running in the background.
+
+Second, if you're compiling a C application rather than a library, you can use [`-s
+PROXY_TO_PTHREAD`](https://emsettings.surma.technology/#PROXY_TO_PTHREAD) option, which will offload
+the main application code to a separate thread in addition to any nested threads created by the
+application itself. This way, main code can block safely at any time without freezing the UI.
+Incidentally, when using this option, you don't have to precreate the thread pool either—instead,
+Emscripten can leverage the main thread for creating new underlying Workers, and then block the
+helper thread in `pthread_join` without deadlocking.
+
+Third, if you're working on a library and still need to block, you can create your own Worker,
+import the Emscripten-generated code and expose it with
+[Comlink](https://github.com/GoogleChromeLabs/comlink) to the main thread. Main thread will be able
+to invoke any exported methods as asynchronous functions, and that way will also avoid blocking the
+UI.
+
+In a simple application such as the previous example  `-s PROXY_TO_PTHREAD` is the best option:
+
+```shell
+emcc -pthread -s PROXY_TO_PTHREAD example.c -o example.js
+```
+
+### C++
+
+All the same caveats and logic apply in the same way to C++. The only new thing you gain is access
+to higher-level APIs like [`std::thread`](https://en.cppreference.com/w/cpp/thread/thread) and
+[`std::async`](https://en.cppreference.com/w/cpp/thread/async), which use the previously discussed
+`pthread` library under the hood.
+
+So the example above can be rewritten in more idiomatic C++ like this:
+
+```cpp/1,9,14
+#include <iostream>
+#include <thread>
+#include <chrono>
+
+int main()
+{
+    puts("Before the thread");
+
+    int arg = 42;
+    std::thread thread([&]() {
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+        std::cout << "Inside the thread: " << arg << std::endl;
+    });
+
+    thread.join();
+
+    std::cout << "After the thread" << std::endl;
+
+    return 0;
+}
+```
+
+When compiled and executed with similar parameters, it will behave in the same way as the C example:
+
+```shell
+emcc -std=c++11 -pthread -s PROXY_TO_PTHREAD example.cpp -o example.js
+```
+
+Output:
+
+```text
+Before the thread
+Inside the thread: 42
+Pthread 0xc06190 exited.
+After the thread
+Proxied main thread 0xa05c18 finished with return code 0. EXIT_RUNTIME=0 set, so keeping
+main thread alive for asynchronous event operations.
+Pthread 0xa05c18 exited.
+```
+
+### Rust
+
+Unlike Emscripten, Rust doesn't have a specialised end-to-end Web target, but instead provides a
+generic `wasm32-unknown-unknown` target for generic WebAssembly output.
+
+If Wasm is intended to be used in a Web environment, any interaction with JavaScript APIs is left to
```

Comment:
```suggestion
If Wasm is intended to be used in a web environment, any interaction with JavaScript APIs is left to
```

Reply:
Looks like this comment was left on outdated text, it already has "web" lowercase in latest version.